bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -101,28 +105,34 @@ public BigtableDataGrpcClient(
     this.channel = channel;
     this.executorService = executorService;
     this.retryOptions = retryOptions;
+    setClientCallService(ClientCallService.DEFAULT);
+  }
 
-    this.sampleRowKeysAsync = BigtableAsyncUtilities.createSampleRowKeyAsyncReader(this.channel);
-    this.readRowsAsync = BigtableAsyncUtilities.createRowKeyAysncReader(this.channel);
+  @VisibleForTesting
+  void setClientCallService(ClientCallService clientCallService) {
```

Comment:
Can you make this part of constructor?

public BigtableDataGrpcClient(
       Channel channel,
       ExecutorService executorService,
       RetryOptions retryOptions) {
    this(channel, executorService, retryOptions, ClientCallService.DEFAULT);
}